src/routes/ViewAllLogs/ViewAllLogs.js

```diff
@@ -159,11 +159,13 @@ class ViewAllLogs extends Component {
         lastName: item.runBy.lastName,
       }))
       .sort((userA, userB) => {
-        if (userA.firstName.localeCompare(userB.firstName) === 0) {
+        const name = userA.firstName || userA.lastName;
+
+        if (userA.firstName?.localeCompare(userB.firstName) === 0) {
           return userA.lastName.localeCompare(userB.lastName);
         }
 
-        return userA.firstName.localeCompare(userB.firstName);
+        return name.localeCompare(userB.firstName);
```

Comment:
What happens when `userA` has `firstName` but `userB.firstName` === `undefined`?

Reply:
Then it should compare value with undefined and sort it, does it seems wrong behavior? Do you propose to have nameA and nameB instead to be sure that we cath all cases?

Reply:
Done